python/GafferUSDUI/USDShaderUI.py

```diff
@@ -190,12 +202,17 @@ def __presetValues( plug ) :
 	property = __primProperty( plug )
 	if property :
 		allowedTokens = property.GetMetadata( "allowedTokens" )
-		return IECore.StringVectorData( allowedTokens ) if allowedTokens else None
+		if allowedTokens and isinstance( plug, Gaffer.IntPlug ) :
+			return IECore.IntVectorData( [ i for i in range( len( allowedTokens ) ) ] )
+		else :
+			return IECore.StringVectorData( allowedTokens ) if allowedTokens else None
```

Comment:
Could you give me an example of a shader where this is necessary? I want to refresh my memory and check that this isn't a symptom of a problem elsewhere.

Reply:
That one I didn't hit in my testing of MaterialX nodes but the check after it is the one which works for me, so this one might be redundant - the IntPlug check should keep it harmless to leave in though...

Reply:
Yep, but what shader can I use to see the problem? I want to check that the problem isn't in the way we're loading the shader rather than in the UI...

Reply:
I think I was checking with `gltf_surface` with the alpha mode plug but there were other shaders I was coming across that needed it, I'll get back to you on a few more tomorrow. I did see from the mtlx schema that they register as integer and not token so I think the loading part was correct (I think token/string is a better approach if that enum list changes though).